source/connection.cc

```diff
@@ -209,6 +211,21 @@ RestClient::Connection::SetKeyPath(const std::string& keyPath) {
   this->keyPath = keyPath;
 }
 
+void
+RestClient::Connection::SetProxy(const std::string& uriProxy) {
+  if (uriProxy.empty())
+    return;
+
+  std::string uriProxyUpper = uriProxy;
+  std::transform(uriProxyUpper.begin(), uriProxyUpper.end(),
+    uriProxyUpper.begin(), ::toupper);
+
+  if (uriProxyUpper.compare(0, 4, "HTTP") != 0)
+    this->uriProxy = "http://" + uriProxy;
+  else
+    this->uriProxy = uriProxy;
```

Comment:
this line seems to not be covered by a unit test (see [here](https://coveralls.io/builds/9306979/source?filename=source%2Fconnection.cc)). Do you know of an https proxy you could use to test this behaviour?